dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLlvm.java

```diff
@@ -376,21 +382,40 @@ public Expression visitCallInst(CallInstContext ctx) {
                 getOrNewRegister(currentRegisterName, returnType);
 
         if (ctx.inlineAsm() != null) {
+            String asmCode = ctx.inlineAsm().inlineAsmBody().getText();
             // see https://llvm.org/docs/LangRef.html#inline-assembler-expressions
             //FIXME ignore side effects of inline assembly
-            CharStream charStream = CharStreams.fromString(ctx.inlineAsm().inlineAsmBody().getText());
-            ParserInlineAsm parser = new ParserInlineAsm(function, resultRegister, arguments);
-            List<Event> events = new ArrayList<>();
-            try{
-                events = parser.parse(charStream);
-            } catch (ParsingException e){
-                logger.warn("Found unrecognized token for inline assembly : '{}'. Setting non deterministic value ", e.getMessage());
-                if(resultRegister != null){
-                    events.add(EventFactory.newLocal(resultRegister, program.newConstant(resultRegister.getType())));
+            final List<ParserAsm> parsers = List.of(
+                    new ParserArm(function, resultRegister, arguments),
+                    new ParserRISCV(function, resultRegister, arguments),
+                    new ParserPPC(function, resultRegister, arguments),
+                    new ParserX86(function, resultRegister, arguments)
+            );
+            Optional<List<Event>> events = Optional.empty();
+            for(ParserAsm parser : parsers){
+                // we have to generate the stream each time as the parser consumes it
+                CharStream charStream = CharStreams.fromString(asmCode);
+                try {
+                    events = tryParse(parser,charStream);    
+                } catch (ProgramProcessingException e) {
+                    logger.warn("Support for inline assembly instruction '{}' is not available for parser '{}'. Setting non deterministic value ", e.getMessage(), parser.getClass().getSimpleName());
+                    if(resultRegister != null){
+                        Event nonDeterministicValue = EventFactory.newLocal(resultRegister, program.newConstant(resultRegister.getType()));
```

Comment:
You should use `EventFactory.Svcomp.newNonDetChoice(Register)`.
I know the previous code didn't do it either, but it was also wrong :).

PS. I need to move this event into the standard namespace; it has nothing to do with Svcomp (anymore).